lib/mac/bindings.js

```diff
@@ -149,110 +149,123 @@ blenoBindings.on('kCBMsgId17', function(args) {
   this.emit('advertisingStop');
 });
 
-blenoBindings.setServices = function(services) {
-  this.sendCBMsg(12, null); // remove all services
-
-  services = services || [];
-  var attributeId = 1;
-
-  this._attributes = [];
-  this._setServicesError = undefined;
-
-  if (services.length) {
-    for (var i = 0; i < services.length; i++) {
-      var service = services[i];
-
-      var arg = {
-        kCBMsgArgAttributeID: attributeId,
-        kCBMsgArgAttributeIDs: [],
-        kCBMsgArgCharacteristics: [],
-        kCBMsgArgType: 1, // 1 => primary, 0 => included
-        kCBMsgArgUUID: new Buffer(service.uuid, 'hex')
-      };
-
-      this._attributes[attributeId] = service;
+blenoBindings.addService = function AddGATTService(service) {
+  var attributeId = this._attributes.length ? this._attributes.length - 1 : 1;
```

Comment:
I think we need a strategy to manage attribute id's when adding and removing, to make sure there are no overlaps ....